src/components/Onboarding/utils.js

```diff
@@ -15,3 +15,9 @@ export const nicelyFormatWalletProvider = provider => {
       return 'your wallet';
   }
 };
+
+export const ensureWalletLink = (hotWallet, coldWallet) => {
+  if (coldWallet && !hotWallet) hotWallet = coldWallet.proxy.linkedAccount;
+  if (hotWallet && !coldWallet) coldWallet = hotWallet.proxy.linkedAccount;
+  return { hotWallet, coldWallet };
+};
```

Comment:
A little more investigation shows that we can catch the bad app state (no `hotWallet` available) here, and provide it with from the linked account property. This way we don't have to mess around with the base layout at all.

Reply:
i love it when a diff gets smaller and smaller 👌

Reply:
agreed!